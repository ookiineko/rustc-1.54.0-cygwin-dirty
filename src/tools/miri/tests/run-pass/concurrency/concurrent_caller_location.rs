// ignore-windows: Concurrency on Windows is not supported yet.

use std::thread::spawn;
use std::panic::Location;

fn initialize() {
    let _ignore = initialize_inner();
}

fn initialize_inner() -> &'static Location<'static> {
    Location::caller()
}

fn main() {
    let j1 = spawn(initialize);
    let j2 = spawn(initialize);
    j1.join().unwrap();
    j2.join().unwrap();
}
