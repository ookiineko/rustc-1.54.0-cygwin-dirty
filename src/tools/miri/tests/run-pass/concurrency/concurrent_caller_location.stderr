warning: thread support is experimental and incomplete: weak memory effects are not emulated.

